microsoft/testsuites/network/networksettings.py

```diff
@@ -494,6 +506,86 @@ def validate_device_msg_level_change(self, node: Node, log: Logger) -> None:
                 f" succeed. Current value is {reverted_settings.msg_level_number}",
             ).is_equal_to(int(original_msg_level_number, 16))
 
+    @TestCaseMetadata(
+        description="""
+            This test case verifies that device statistics can be captured.
+
+            Steps:
+            1. Get all the device's statistics.
+            2. Validate device statistics lists per cpu statistics as well.
+        """,
+        priority=2,
+    )
+    def validate_device_statistics(self, environment: Environment, log: Logger) -> None:
+        client_iperf3_log = "iperfResults.log"
+        server_node = cast(RemoteNode, environment.nodes[0])
+        client_node = cast(RemoteNode, environment.nodes[1])
+        ethtool = client_node.tools[Ethtool]
+
+        try:
+            devices_statistics = ethtool.get_all_device_statistics()
+        except UnsupportedOperationException as identifier:
+            raise SkippedException(identifier)
+
+        for stats in devices_statistics:
+            assert_that(stats, "Statistics are empty").is_not_empty()
+
+        # preparation work before launch iperf3
+        stop_firewall(environment)
+
+        # run iperf3 on server side and client side
+        # iperfResults.log stored client side log
+        source_iperf3 = server_node.tools[Iperf3]
+        dest_iperf3 = client_node.tools[Iperf3]
+        source_iperf3.run_as_server()
+        dest_iperf3.run_as_client_async(
+            server_node.internal_address, client_iperf3_log, 1800, parallel_number=64
```

Comment:
1. Will it run 1800 seconds? it's too long to a functional test case. Try default value 10 seconds, it should be enough.
2. To get it more stable, do not use async here. Use synchronous method.